packages/react-vapor/src/components/input/Input.tsx

```diff
@@ -243,3 +248,24 @@ export class Input extends React.Component<IInputProps, IInputComponentState> {
         );
     }
 }
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IInputProps): IInputStateProps => {
+    const input = InputSelectors.getInput(state, {id: ownProps.id});
+    return {
+        valid: input && input.valid,
+        value: input && input.value,
+        disabled: input && input.disabled,
+    };
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IInputProps): IInputDispatchProps => ({
+    onRender: (value: string = '', valid = true, disabled = false) =>
+        dispatch(addInput(ownProps.id, value, valid, disabled)),
+    onDestroy: () => dispatch(removeInput(ownProps.id)),
+    onChange: (value: string, valid = true) => {
+        dispatch(changeInputValue(ownProps.id, value, valid));
+        ownProps.changeDirtyState?.(value);
+    },
+});
+
+export const InputConnected = connect(mapStateToProps, mapDispatchToProps, ReduxUtils.mergeProps)(Input);
```

Comment:
I was wondering, what `ReduxUtils.mergeProps` doing exactly? Is it still relevant to have it?

Reply:
It inverts the order of props overriding. Without it the state props override the own props. With it the own props override the state props. Even if its not ideal, since it was already that way, I don't want to remove it in case something somewhere relies on this behaviour.